### PR TITLE
Update setInterval

### PR DESCRIPTION
**Pull request by samuelds**
Add args for callback for setInterval

### CODE REVIEW

File: data/shared/citizen/scripting/v8/timer.js [hunk 1]
```diff
@@ -85,10 +85,18 @@
         animationFrames[animationFrames.length] = callback;
     }
 
-    function setInterval(callback, interval) {
+    function setInterval(callback, interval, ...argsForCallback) {
         const id = nextTimerId();
 
-        setTimer(id, callback, interval);
+	setTimer(
```
Comment (Hellslicer) [inline data/shared/citizen/scripting/v8/timer.js]: Bad indentation
```suggestion
        setTimer(
```

File: data/shared/citizen/scripting/v8/timer.js [hunk 2]
```diff
@@ -85,10 +85,18 @@
...
+            id,
+            function() {
+                try {
+                    callback(...argsForCallback);
+                }
```
Comment (Hellslicer) [inline data/shared/citizen/scripting/v8/timer.js]: Missing catch or finally after try
```suggestion
                } catch { }
```

File: data/shared/citizen/scripting/v8/timer.js [hunk 3]
```diff
@@ -85,10 +85,18 @@
         animationFrames[animationFrames.length] = callback;
     }
 
-    function setInterval(callback, interval) {
+    function setInterval(callback, interval, ...argsForCallback) {
```
Comment (thers) [inline data/shared/citizen/scripting/v8/timer.js]: It would be great if you'd also add support for the args passing to the `setTimeout` method as well.

Comment (samuelds) [reply data/shared/citizen/scripting/v8/timer.js]: That's why I added argsForCallback

Comment (samuelds) [reply data/shared/citizen/scripting/v8/timer.js]: setTimeout has already:

```js
test(arg) {
   console.log(arg); // Print exemple
}

setTimeout(test, 100, "exemple");
```

Comment (thers) [reply data/shared/citizen/scripting/v8/timer.js]: Aww, yeah, you're right, it's already there, my bad.

File: data/shared/citizen/scripting/v8/timer.js [hunk 4]
```diff
@@ -85,10 +85,18 @@
...
         const id = nextTimerId();
 
-        setTimer(id, callback, interval);
+        setTimer(
+            id,
+            function() {
+                try {
+                    callback(...argsForCallback);
+                } catch { }
```
Comment (thers) [inline data/shared/citizen/scripting/v8/timer.js]: This is bad. If something will throw an error inside `callback` we won't ever notice it failing in console. I'd better write `setTimer(id, callback.bind(null, ...argsForCallback), interval);`

Comment (samuelds) [reply data/shared/citizen/scripting/v8/timer.js]: try and catch is there for that. Just add a print in catch.

Comment (thers) [reply data/shared/citizen/scripting/v8/timer.js]: And you will do what already is there. See https://github.com/citizenfx/fivem/blob/master/data/shared/citizen/scripting/v8/timer.js#L127-L131

Comment (thers) [reply data/shared/citizen/scripting/v8/timer.js]: Thing that process every timer callback is already doing that, why doing it here again?

Comment (samuelds) [reply data/shared/citizen/scripting/v8/timer.js]: I did not pay attention, thx :)

**Pull request merged**